packages/core-mobile/app/hooks/earn/useNodes.ts

```diff
@@ -1,13 +1,24 @@
-import { useQuery } from '@tanstack/react-query'
+import { pvm } from '@avalabs/avalanchejs'
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { useAvalancheXpProvider } from 'hooks/networks/networkProviderHooks'
 import { useSelector } from 'react-redux'
 import EarnService from 'services/earn/EarnService'
 import { selectIsDeveloperMode } from 'store/settings/advanced'
 
-export const useNodes = () => {
+export const useNodes = (): UseQueryResult<
+  pvm.GetCurrentValidatorsResponse,
+  Error
+> => {
   const isDeveloperMode = useSelector(selectIsDeveloperMode)
+  const provider = useAvalancheXpProvider(isDeveloperMode)
 
   return useQuery({
-    queryKey: ['nodes', isDeveloperMode],
-    queryFn: () => EarnService.getCurrentValidators(isDeveloperMode)
+    queryKey: ['nodes', provider],
```

Comment:
it's not recommended to include object as part of queryKey array. can you use something else instead like provider's networkId/url?